### PR TITLE
QOLOE 120 Enhancements

### PR DESCRIPTION
**Pull request by MitchAnderson93**
**Current state**
-------------------------------

Dont merge - using as delta

Please review against:
https://ssq-qol.atlassian.net/browse/QOLOE-120 as per enhancement discussion in comments.

**Contains:**
> - Mobile CTA gap (appears in dropdown) / requires additional JSON for navbar.data file 
> - Changes to tabindex gap (CTA's) desktop 
> - Fix for some firefox tabbing issues 
> - Scroll overlay on large site (background should fix)
> - Header stories for external branding types have been updated and include colour palette to test with 
> - Fix for site search click outside / original scripting was playing up when switching story/colour mode

**Adding:**
> - Add interaction tests for search behaviour (storybook)
> - Popper position on CTA's (other way around)

**Reminders**
-------------------------------

**Known defects:**
> - Missing animations (desktop drop down/mobile slide out)
> - Border (none) on final row items in mega menu (desktop) - hence why I made two helper's

**Still to do (non func):**
> - Fix npm audit errors
> - Clean up/review JS/SCSS conventions to improve readability where needed
>- Thorough testing required for mobile menu (popper instance positioning in CSS on mobile/edge cases) 
> - Enhance storybook controls to showcase all element settings (including wrapping each element in light/dark/dark-alt so users have better control of the colour variation settings in metadata

### CODE REVIEW

File: src/components/bs5/header/header.scss [hunk 1]
```diff
@@ -481,7 +481,6 @@
 				left: 0;
 				right: 0;
 				padding: 1rem;
-				margin-top: 1rem;
```
Comment (MitchAnderson93) [inline src/components/bs5/header/header.scss:484]: _unused_

**Review by duttonw — APPROVED** [review]

File: src/components/bs5/navbar/navbar.data.json [hunk 2]
```diff
@@ -141,6 +141,126 @@
                 "icon": "",
                 "target_url": "index.html"
             }
+        ],
+        "CTA": [
```
Comment (MitchAnderson93) [inline src/components/bs5/navbar/navbar.data.json:145]: CTA's for mobile view - appear the same as standard nav links despite design (I think the design is wrong)

@stvp-qld

File: src/components/bs5/navbar/navbar.functions.js [hunk 3]
```diff
@@ -67,10 +69,12 @@ export function initializeNavbar() {
     // Overlay show/hide events
     navbarCollapse.addEventListener('show.bs.collapse', function () {
       overlay.classList.add('show'); // Show the overlay
+      body.style.overflow = 'hidden'; // Prevent background scroll
     });
 
     navbarCollapse.addEventListener('hide.bs.collapse', function () {
       overlay.classList.remove('show'); // Hide the overlay
+      body.style.overflow = ''; // Reset body positioning
```
Comment (MitchAnderson93) [inline src/components/bs5/navbar/navbar.functions.js:77]: @stvp-qld overflow:hidden as per health.qld.gov.au when mobile overlay is 'show' with states reset

File: src/components/bs5/navbar/navbar.hbs [hunk 4]
```diff
@@ -17,146 +17,231 @@
...
-                                            </svg>
-                                        </a>
-                                        {{#if description}}
-                                            <p class="dropdown-item__description">{{description}}</p>
-                                        {{/if}}
-                                    </li>
-                                {{/each }} 
-                                {{#if dropdown_options.view_value}}
+                                    {{#each dropdown_options.dropdown_config.groups}}
+                                        {{#isType ../dropdown_options.dropdown_type "list"}}
```
Comment (MitchAnderson93) [inline src/components/bs5/navbar/navbar.hbs:205]: @stvp-qld should be the same JSON in header (had to double up in nav JSON)

**Comment by duttonw** [comment]
Hi @MitchAnderson93 ,

Can you ``git pull origin develop --rebase`` and remove the merge's to give this branch a clean history. Unsure what is going on with our pr's but we are getting deltas that should not exist (i.e. maybe some squash's occurred with merge history that is confusing things)

**Review by duttonw — APPROVED** [review]

**Review by duttonw — APPROVED** [review]
Still require another's approval

**Review by ghazal-salehi — APPROVED** [review]

**Review by stvp-qld — APPROVED** [review]

File: src/components/bs5/header/header.stories.js [hunk 5]
```diff
@@ -1,38 +1,170 @@
...
+import {
+  masterbrand_variant,
+  subbrand_variant,
+  cobrand_variant,
+  endorsed_variant,
+  standalone_variant,
+} from './header.data.json';
+
+const sample_argtypes = {
+  '--qld-color-default-color-light-background-default': '#ffffff',
```
[the remark this answers is not in the document]
Comment (MitchAnderson93) [reply src/components/bs5/header/header.stories.js:24]: Just had a look into vape truths campaign (https://vapingexposed.initiatives.qld.gov.au/whats-really-in-vapes) and health are using this convention:

--QLD-color-light__site-title: #000000; //which is more succinct 

@re-becca what do you think?

**Pull request merged**